lib/utils/pulsar_functions.h

```diff
@@ -0,0 +1,36 @@
+#ifndef PULSAR_FUNCTIONS_H
+#define PULSAR_FUNCTIONS_H
+
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct PSRHeader {
+    uint32_t seconds : 30;
+    uint32_t legacy : 1;
+    uint32_t invalid : 1;
+    uint32_t data_frame : 24;
+    uint32_t ref_epoch : 6;
+    uint32_t unused : 2;
+    uint32_t frame_len : 24;
+    uint32_t log_num_chan : 5;
+    uint32_t vdif_version : 3;
+    uint32_t station_id : 16;
+    uint32_t thread_id : 10;
+    uint32_t bits_depth : 5;
+    uint32_t data_type : 1;
+    uint32_t eud1 : 24;
+    uint32_t edv : 8;
+    uint32_t eud2 : 32;
+    uint32_t eud3 : 32;
+    uint32_t eud4 : 32;
```

Comment:
Since this is no longer "VDIF" you could rename/resize these to match the data going into them.  If you don't want to do that just yet, it would be worth putting comments into here as to what data they contain.   Since people are more likely to look here than code that sets the values to figure out what they are.

Reply:
good point; i've added some small doc-like strings to describe their use.